lib/Demangling/Demangler.cpp

```diff
@@ -185,6 +185,7 @@ int swift::Demangle::getManglingPrefixLength(llvm::StringRef mangledName) {
     /*Swift 4*/   "_T0",
     /*Swift 4.x*/ "$S", "_$S",
     /*Swift 5+*/  "$s", "_$s",
+    /*Swift 5+ Embedded Swift*/  "$e", "_$e",
```

Comment:
Can you add a function here that returns the Mangling Flavor of a string, and declare it on `Demangle.h` as well?

Something like:
```
std::optional<Mangle::ManglingFlavor> getManglingFlavor(llvm::StringRef mangledName) {
  if (!isSwiftSymbol(mangledName))
    return {};

  if (mangledName.starts_with("$e") || mangledName.starts_with("_$e"))
    return Mangle::ManglingFlavor::Embedded;
  return Mangle::ManglingFlavor::Default;
}
```